From e94608eac69d78ef44dde8c2474f62f0da54f9c8 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 16 Jul 2014 20:42:28 -0400
Subject: [md] dm-cache: fail migrations in the do_worker error path

Message-id: <1405543364-20901-25-git-send-email-snitzer@redhat.com>
Patchwork-id: 85915
O-Subject: [RHEL7.1 PATCH 24/40] dm cache: fail migrations in the do_worker error path
Bugzilla: 1117872
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>
RH-Acked-by: Joe Thornber <thornber@redhat.com>

BZ: 1117872

Upstream linux-dm.git commit fbffa67d00106a0fc8950c0d012f094866ee702c
Author: Joe Thornber <ejt@redhat.com>
Date:   Tue Jun 24 15:36:58 2014 -0400

    dm cache: fail migrations in the do_worker error path

    Signed-off-by: Joe Thornber <ejt@redhat.com>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-cache-target.c b/drivers/md/dm-cache-target.c
index e628968..32e224a 100644
--- a/drivers/md/dm-cache-target.c
+++ b/drivers/md/dm-cache-target.c
@@ -1715,6 +1715,7 @@ static void do_worker(struct work_struct *ws)
 
   if (commit_if_needed(cache)) {
    process_deferred_flush_bios(cache, false);
+   process_migrations(cache, &cache->need_commit_migrations, migration_failure);
 
    /*
     * FIXME: rollback metadata or just go into a
-- 
1.7.1